torch/fx/passes/infra/partitioner.py

```diff
@@ -148,26 +142,6 @@ def merge_single_node(node: Node, id: int):
                     # this is a no-op
                     maybe_merge_partition(self_id, other_id)
 
-        # post processing to re-assign "getitem" nodes into upstream partition
```

Comment:
nitpick, should we remove this part?

This was supposed to fuse getitem to producer, which is a fusion, before we have `getitem` disabled in the line you commented above. With that logic removed, I believe this section would remove getitem node at the beginning of the fusion, if it comes from an unfused node.

Reply:
Keeping this code works with the example I added to the PR description, but I was hitting the assert in `merge_single_node` in one benchmark model. I should extract that failing graph portion for testing.

Reply:
Got ya. merge_single_node might be asserting too aggressively as well.

https://github.com/pytorch/pytorch/blob/8393213a04542b1be58170a79afe5312f1871c72/torch/fx/passes/infra/partitioner.py#L108

This assert should be removed along with you update on `getitem` logic.

Reply:
Keeping this code `python -m pytest test/test_fx_passes.py -k "test_partitioner_fn_" -vvv` fails with the assert
```py
Traceback (most recent call last):
  File "/home/iyashchuk/dev/pytorch/master/test/test_fx_passes.py", line 224, in test_partitioner
    partitions = partitioner.propose_partitions()
  File "/home/iyashchuk/dev/pytorch/master/torch/fx/passes/infra/partitioner.py", line 162, in propose_partitions
    merge_single_node(node, id)
  File "/home/iyashchuk/dev/pytorch/master/torch/fx/passes/infra/partitioner.py", line 108, in merge_single_node
    assert node not in assignment
AssertionError
```
Removing this code the test still fails (as in CI) but the failure is expected now:
```py
____ TestFXGraphPasses.test_partitioner_fn_<function TestPartitionFunctions_forward13 at 0x7f08e8fdfd00>_expected_partition_[['add_2', 'add_1', 'add']] ____
Traceback (most recent call last):
  File "/home/iyashchuk/dev/pytorch/master/test/test_fx_passes.py", line 229, in test_partitioner
    assert set(partitions_name[i]) == set(expected_partition[i])
AssertionError: assert {'add', 'getitem_2', 'getitem', 'getitem_1', 'getitem_3', 'add_2', 'add_1'} == {'add_2', 'add_1', 'add'}
  Extra items in the left set:
  'getitem_2'
  'getitem_3'
  'getitem_1'
  'getitem'
  Full diff:
  - {'add_2', 'add_1', 'add'}
  + {'add', 'getitem_2', 'getitem', 'getitem_1', 'getitem_3', 'add_2', 'add_1'}
```
The failing test was added in `https://github.com/pytorch/pytorch/pull/86713` 😉

Reply:
I was suggesting keeping this code and remove the assert.

Reply:
Oops, that's surprising to me. :laughing:

How can we have `getitem` spanning acorss multiple partition? I think there's just a small bug somewhere. Looks like it's attempting to clear `getitem` from the original partition... But somehow that wasn't done right. I can take a quick look at this afterwards.